(err-std_mis-notebook)=
# L'errore standard della misurazione 

In [1]:
source("../_common.R")
suppressWarnings(suppressMessages(library("psychometric")))

I coefficienti di affidabilità che abbiamo discusso nel capitolo precedente rappresentano una misura proporzionale della varianza osservata di un test che è attribuibile alla varianza reale. Questi coefficienti sono fondamentali per confrontare l'affidabilità dei punteggi ottenuti da diverse procedure di valutazione. In generale, preferiremo selezionare il test che produce i punteggi con la migliore affidabilità. Tuttavia, una volta scelto il test, il nostro focus si sposta sull'interpretazione dei punteggi.

## Errore Standard di Misurazione (SEM)

L'Errore Standard di Misurazione (SEM) diventa una statistica più pratica quando l'attenzione è rivolta all'interpretazione dei punteggi di un test. Il SEM è definito come la deviazione standard della distribuzione dei punteggi che un individuo otterrebbe se fosse sottoposto a un numero infinito di forme parallele del test, costituite da item campionati casualmente dallo stesso dominio di contenuto.

Per comprendere meglio, immaginiamo di creare un numero infinito di forme parallele di un test e di far svolgere queste forme alla stessa persona, senza che vi siano effetti di trasferimento. La presenza dell'errore di misurazione impedirebbe alla persona di ottenere sempre lo stesso punteggio. Anche se ogni test rappresenta ugualmente bene il dominio di contenuto, il candidato potrebbe ottenere risultati migliori in alcuni test e peggiori in altri, semplicemente a causa di errori casuali (ad esempio, la fortuna nel conoscere le risposte agli item selezionati per una versione del test ma non per un'altra). Prendendo i punteggi ottenuti in tutti questi test, si otterrebbe una distribuzione di punteggi. La media di questa distribuzione rappresenta il punteggio vero (T) dell'individuo, mentre il SEM è la deviazione standard di questa distribuzione di punteggi di errore.

Ovviamente, non è possibile attuare questi procedimenti nella realtà, quindi dobbiamo stimare il SEM utilizzando le informazioni disponibili. Esamineremo qui l'approccio utilizzato dalla Teoria Classica dei Test (CTT) per raggiungere questo obiettivo. 

## Stima di SEM

Secondo Lord (1968), l'errore $E = X - T$ rappresenta la variabile aleatoria di interesse primario nella CTT. L'obiettivo della CTT è stimare il punteggio vero di ogni rispondente e confrontare le stime ottenute per rispondenti diversi. La grandezza dell'errore $E$ fornisce informazioni essenziali in questo contesto. La discrepanza tra il punteggio osservato e il punteggio vero può essere misurata utilizzando la deviazione standard degli errori $E$, conosciuta appunto come "Errore Standard della Misurazione" o SEM. Il SEM è quindi lo strumento impiegato dalla CTT per stimare in che misura un punteggio osservato differisce dal punteggio vero.

Nel presente capitolo esploreremo come sia possibile stimare la deviazione standard dell'errore ($\sigma_E$) in un campione di osservazioni. Questo consente di comprendere meglio la precisione dei punteggi ottenuti attraverso un test psicometrico e di interpretare in modo più accurato i risultati.

## L'incertezza della misura 

In base alla CTT, è possibile stimare l'errore standard della misurazione utilizzando una formula che dipende dalla deviazione standard della distribuzione dei punteggi del test e dall'attendibilità del test. Mediante questa formula, è possibile ottenere una stima dell'errore standard associato a un singolo punteggio, il quale indica quanto il punteggio osservato può variare rispetto al vero punteggio di un individuo:

$$
\sigma_E = \sigma_X \sqrt{1 -\rho_{XX^\prime}},
$$ (eq-err-stnd-mis)

dove $\sigma_X$ rappresenta la deviazione standard dei punteggi ottenuti da un campione di soggetti e $\rho_{XX^\prime}$ è il coefficiente di attendibilità. Attraverso questo calcolo, si ottiene l'errore standard della misurazione sottraendo l'attendibilità del test da 1, quindi calcolando la radice quadrata del risultato e moltiplicandolo per la deviazione standard dei punteggi del test.

La logica alla base dell'errore standard della misurazione si fonda sull'assunzione che se una persona dovesse sostenere numerosi test equivalenti, i punteggi ottenuti seguirebbero una distribuzione normale con il vero punteggio dell'individuo come media. In altre parole, possiamo immaginare che l'individuo affronti ripetutamente versioni identiche del test, in circostanze simili e senza ricordare le risposte precedenti. In tale contesto ipotetico, l'errore standard della misurazione rappresenterebbe la deviazione standard tra queste misurazioni ripetute.

La formula sopra indicata evidenzia come l'errore standard della misurazione ($\sigma_E$) sia strettamente correlato all'attendibilità del test: all'aumentare dell'attendibilità del test, l'errore standard della misurazione diminuisce. Se l'attendibilità del test si avvicina a 0, l'errore standard della misurazione tende a diventare uguale alla deviazione standard dei punteggi osservati del test. In contrasto, se l'attendibilità del test raggiunge 1, l'errore standard della misurazione si riduce a zero: in una situazione di perfetta affidabilità, in cui non vi è alcun errore di misurazione, $\sigma_E$ assume valore zero.

### Interpretazione

Secondo la prospettiva di McDonald, il termine di errore ($E$) segue una "propensity distribution", che riflette le fluttuazioni casuali nel tempo di un individuo nel rispondere al test. Queste fluttuazioni possono derivare da variabili come l'umore, la motivazione e altri fattori. L'errore standard di misura fornisce una stima della deviazione standard di tali punteggi, rappresentando la dispersione attesa dei punteggi che un singolo individuo otterrebbe in un numero infinito di somministrazioni del test (o di test paralleli) sotto identiche condizioni, nel caso in cui il vero punteggio rimanesse costante.

D'altro canto, la CTT assume che i punteggi ottenuti da un individuo in infinite somministrazioni del test, nelle stesse condizioni, seguano una distribuzione normale centrata sul vero valore. L'errore standard di misura diviene, quindi, la stima della deviazione standard di questa distribuzione teorica di punteggi. Maggiore è l'errore standard di misura, maggiore è l'incertezza introdotta nel processo di utilizzo del test per valutare l'abilità latente del rispondente.

Il coefficiente di attendibilità, la varianza dell'errore e l'errore standard di misura sono tutti indicatori diretti o indiretti della precisione del test. Tuttavia, ciascuno di essi fornisce un'informazione specifica sul livello di precisione del test:

-   L'errore standard di misura permette di fare inferenze sulla precisione del singolo punteggio osservato di un partecipante, mentre il coefficiente di attendibilità non ha una rappresentazione diretta in questo senso.
-   L'errore standard di misura è espresso nella stessa unità di misura del punteggio osservato, a differenza della varianza di $E$, che è espressa in termini del quadrato del punteggio osservato.
-   L'attendibilità rappresenta un rapporto tra varianze ed è quindi un indice adimensionale.

**Esempio 1.** Consideriamo un esempio in cui un test di intelligenza fornisce un punteggio medio di 100 con una deviazione standard di 15. Supponiamo inoltre che l'attendibilità di questo test sia pari a 0.73. Vogliamo calcolare l'errore standard della misurazione.

Utilizzando la formula dell'errore standard della misurazione, otteniamo:

$$
\begin{equation}
\begin{aligned}
\sigma_E &= \sigma_X \sqrt{1 -\rho_{XX^\prime}} \notag\\
&= 15 \sqrt{1 - 0.73} \notag\\
&= 7.79.\notag
\end{aligned}
\end{equation}
$$

Il valore 7.79 rappresenta l'errore standard atteso nei punteggi ottenuti da un singolo individuo se il test fosse somministrato più volte sotto identiche condizioni. In altre parole, ci aspettiamo che i punteggi variino in media di circa 8 punti tra diverse somministrazioni del test.

Inoltre, possiamo utilizzare l'errore standard della misurazione per calcolare un intervallo di confidenza intorno al vero punteggio del rispondente. Utilizzando la proprietà della distribuzione gaussiana, possiamo stimare che il 95% dei punteggi ottenuti da ripetute somministrazioni del test si troveranno nell'intervallo:

$$
\text{punteggio vero del rispondente} \pm 1.96 \cdot \text{errore standard della misurazione}. 
$$

Nel nostro caso, questo intervallo sarebbe pari a $2 \cdot 1.96 \cdot 7.79 = 30.54$ punti. Quindi, ci aspettiamo che i punteggi del QI di un singolo rispondente varino all'interno di un intervallo di 30 punti se il test fosse somministrato molte volte sotto le stesse condizioni.

Questo esempio dimostra che se un test ha un'attendibilità di 0.73 e una deviazione standard dei punteggi di 15, la misurazione del test su un singolo individuo risulterebbe poco affidabile a causa dell'ampio errore di misurazione. A titolo di confronto, la Full Scale IQ (FSIQ) della WAIS-IV {cite:p}`wechsler2008wechsler` ha un'attendibilità split-half di 0.98 e un errore standard di misurazione di 2.16.

L'errore standard della misurazione può anche essere calcolato utilizzando la funzione `SE.Means()` del pacchetto `psychometric`.

In [4]:
SE.Meas(15, .73)

[1] 7.794229

**Esempio 2.** Continuando con l'esempio precedente, per gli ipotetici dati riportati
sopra, poniamoci ora la seguente domanda: qual è la probabilità che un
rispondente ottenga un punteggio minore o uguale a 116 nel test, se il
suo punteggio vero è uguale a 120?

Il problema si risolve rendendosi conto che i punteggi del
rispondente si distribuiscono normalmente attorno al punteggio vero di
120, con una deviazione standard uguale a 7.79. Dobbiamo dunque trovare
l'area sottesa alla normale $\mathcal{N}(120, 7.79)$ nell'intervallo
$[-\infty, 116]$. Utilizzando , la soluzione si trova nel modo
seguente:

In [5]:
pnorm(116, 120, 7.79)

[1] 0.3038082

Se la variabile aleatorie corrispondente al punteggio osservato segue
una distribuzione $\mathcal{N}(120, 7.79)$, la probabilità che il
rispondente ottenga un punteggio minore o uguale a 116 è dunque uguale a
0.30.

**Esempio 3.** Sempre per l'esempio discusso, poniamoci ora la seguente domanda: quale intervallo di valori centrato sul punteggio vero contiene, con una probabilità di 0.95, i punteggi che il rispondente otterrebbe in ipotetiche somministrazioni ripetute del
test sotto le stesse identiche condizioni?

Dobbiamo trovare i quantili della distribuzione $\mathcal{N}(120, 7.79)$ a cui sono associate le probabilità di 0.025 e 0.975. La soluzione è dunque data da:

In [6]:
qnorm(c(.025, .975), 120, 7.79)

[1] 104.7319 135.2681

L'intervallo cercato è dunque $[104.7, 135.3]$.

**Esempio 4.** Calcoliamo ora l'errore standard di misurazione utilizzando un campione di dati grezzi. Esamineremo un set di dati discusso da {cite:t}`brown2015confirmatory`. I dati consisteono in 9 indicatori usati per misurare la depressione maggiore così come è definita nel DSM-IV:

- MDD1: depressed mood;
- MDD2: loss of interest in usual activities;
- MDD3: weight/appetite change;
- MDD4: sleep disturbance;
- MDD5: psychomotor agitation/retardation;
- MDD6: fatigue/loss of energy;
- MDD7: feelings of worthlessness/guilt;
- MDD8: concentration difficulties;
- MDD9: thoughts of death/suicidality.

Importiamo i dati:

In [27]:
df <- readRDS(
    here::here("data", "mdd_sex.RDS")
) |>
    dplyr::select(-sex)

Ci sono 750 osservazioni:

In [21]:
dim(df)

[1] 750   9

In [20]:
head(df)

mdd1 mdd2 mdd3 mdd4 mdd5 mdd6 mdd7 mdd8 mdd9
1 5    4    1    6    5    6    5    4    2   
2 5    5    5    5    4    5    4    5    4   
3 4    5    4    2    6    6    0    0    0   
4 5    5    3    3    5    5    6    4    0   
5 5    5    0    5    0    4    6    0    0   
6 6    6    4    6    4    6    5    6    2

Calcoliamo il coefficiente di attendibilità $\alpha$ di Cronbach con la funzione `alpha()` del pacchetto `psych`.

In [40]:
res <- psych::alpha(df)
alpha <- res$total$raw_alpha
alpha

[1] 0.7531505

Calcoliamo un vettore che contiene il punteggio totale del test per ciascun individuo:

In [ ]:
total_score <- rowSums(df)

Infine, troviamo l'errore standard di misurazione:

In [41]:
sd(total_score) * sqrt(1 - alpha)

[1] 5.296432

Confrontiamo il risultato con quello ottenuto con la funzione `SE.Meas()`:

In [42]:
SE.Meas(sd(total_score), alpha)

[1] 5.296432

## Dimostrazione

Esaminiamo ora la derivazione della formula per l'errore standard di misurazione, $\sigma_E = \sigma_X \sqrt{1 - \rho_{XX^\prime}}$. Per arrivare a questa formula, seguiremo due passaggi chiave: innanzitutto, calcoleremo la varianza del vero punteggio e successivamente rappresenteremo il punteggio osservato come la somma della varianza del vero punteggio e la varianza dell'errore.

Iniziamo definendo il coefficiente di attendibilità come $\rho_{XX^\prime} = \frac{\sigma^2_T}{\sigma^2_X}$, in cui $\sigma^2_T$ è la varianza del vero punteggio e $\sigma^2_X$ è la varianza del punteggio osservato. Utilizzando questa definizione, possiamo riscrivere $\sigma^2_T$ come $\sigma^2_T = \rho_{XX^\prime} \sigma^2_X$, considerando che $X$ e $X^\prime$ sono forme parallele di un test e con proprietà simili.

Dato che $\sigma_X = \sigma_{X^\prime}$, possiamo semplificare l'equazione precedente in $\sigma^2_T = \rho_{XX^\prime} \sigma_X \sigma_{X^\prime}$. Inoltre, la covarianza tra $X$ e $X^\prime$ è definita come $\sigma_{XX^\prime} = \rho_{XX^\prime} \sigma_X \sigma_{X^\prime}$. Da qui, possiamo affermare che $\sigma^2_T = \sigma_{XX^\prime}$, dato che la varianza del vero punteggio equivale alla covarianza tra due misurazioni parallele.

Ora, passiamo a calcolare la varianza dell'errore, $\sigma^2_E$. La varianza del punteggio osservato è espressa come $\sigma^2_X = \sigma^2_T + \sigma^2_E$. Utilizzando la definizione di attendibilità, possiamo riscrivere questa equazione come $\sigma^2_X = \rho_{XX^\prime} \sigma^2_X + \sigma^2_E$, da cui otteniamo:

$$
\begin{equation}
\begin{aligned}
\sigma^2_E &= \sigma^2_X - \sigma^2_X \rho_{XX^\prime} \\
&= \sigma^2_X (1 - \rho_{XX^\prime}).
\end{aligned}
\end{equation}
$$

Di conseguenza, la varianza dell'errore di misurazione, $\sigma^2_E$, può essere espressa come il prodotto di due fattori: il primo rappresenta la varianza del punteggio osservato, mentre il secondo equivale a uno meno la correlazione tra le due forme parallele del test ($\rho_{XX^\prime}$). In conclusione, abbiamo calcolato l'incognita $\sigma^2_E$ in termini di due quantità osservabili, $\sigma^2_X$ e $\rho_{XX^\prime}$.

## Relazione tra Affidabilità del Test e Errore Standard di Misurazione (SEM)

Si osserva che, all'aumentare dell'affidabilità di un test, l'Errore Standard di Misurazione (SEM) diminuisce. Questa relazione inversa è coerente con il fatto che il coefficiente di affidabilità riflette la proporzione della varianza dei punteggi osservati dovuta alla varianza dei punteggi veri, e il SEM è una stima dell'errore presente nei punteggi del test. Quindi, maggiore è l'affidabilità dei punteggi di un test, minore è il SEM, e maggior fiducia possiamo avere nella precisione dei punteggi del test. Viceversa, minore è l'affidabilità di un test, maggiore è il SEM, e minore è la nostra fiducia nella precisione dei punteggi del test.

Per esempio, con un coefficiente di affidabilità perfetto pari a 1.0, il SEM sarebbe uguale a 0, indicando l'assenza di errore nella misurazione e che il punteggio ottenuto rappresenta il punteggio vero. Un coefficiente di affidabilità pari a 0, invece, produrrebbe un SEM uguale alla deviazione standard (SD) dei punteggi ottenuti, indicando che tutta la varianza dei punteggi del test è dovuta a errori.

Il SEM è tradizionalmente utilizzato nel calcolo di intervalli o bande intorno ai punteggi osservati, all'interno dei quali ci si aspetta che cada il punteggio vero. Ora passeremo a questa applicazione del SEM.

## Intervallo di Confidenza e Errore Standard di Misurazione (SEM)

L'intervallo di confidenza rappresenta un range di punteggi che include il vero punteggio di un individuo con una probabilità prescritta. Generalmente, utilizziamo il SEM per calcolare gli intervalli di confidenza. Il SEM fornisce informazioni sulla distribuzione dei punteggi osservati intorno ai punteggi veri. 

Ad esempio, se un individuo ha un punteggio vero di 70 in un test con un SEM di 3, ci aspetteremmo che ottenga punteggi tra 67 e 73 due terzi delle volte, a patto che non ci siano cambiamenti nelle prestazioni a causa della ripetizione del test. 

In [3]:
# Definiamo il punteggio vero e il SEM
punteggio_vero <- 70
SEM <- 3

pnorm(73, 70, 3) - pnorm(67, 70, 3)

[1] 0.6826895

Per ottenere un intervallo di confidenza del 95%, determiniamo il numero di deviazioni standard che comprendono il 95% dei punteggi in una distribuzione. Con un punteggio vero di 70 e un SEM di 3, l'intervallo di confidenza del 95% sarebbe 70 ± 3(1.96), ovvero 70 ± 5.88. Quindi, in questa situazione, ci aspetteremmo che il punteggio osservato dell'individuo sia tra 64.12 e 75.88, il 95% delle volte.

In [2]:
# Calcoliamo il valore critico Z per il livello di confidenza del 95%
livello_confidenza <- 0.95
z_critico <- qnorm((1 + livello_confidenza) / 2)

# Calcoliamo l'errore standard dell'intervallo
errore_standard_intervallo <- SEM * z_critico

# Calcoliamo l'intervallo di confidenza
intervallo_confidenza_inf <- punteggio_vero - errore_standard_intervallo
intervallo_confidenza_sup <- punteggio_vero + errore_standard_intervallo

# Stampiamo l'intervallo di confidenza
cat("L'intervallo di confidenza al 95% e' [", intervallo_confidenza_inf, ", ", intervallo_confidenza_sup, "]\n")


L'intervallo di confidenza al 95% e' [ 64.12011 ,  75.87989 ]


### Relazione tra Affidabilità, SEM e Intervalli di Confidenza

È utile notare la relazione tra l'affidabilità di un punteggio di test, il SEM e gli intervalli di confidenza. Ricordiamo che all'aumentare dell'affidabilità dei punteggi, il SEM diminuisce. La stessa relazione esiste tra l'affidabilità dei punteggi di test e gli intervalli di confidenza. Man mano che l'affidabilità dei punteggi di test aumenta (denotando meno errore di misurazione), gli intervalli di confidenza diventano più piccoli (denotando maggiore precisione nella misurazione).

### Vantaggio del SEM e dell'Uso degli Intervalli di Confidenza

Un grande vantaggio del SEM e dell'uso degli intervalli di confidenza è che essi servono a ricordarci che l'errore di misurazione è presente in tutti i punteggi e che dovremmo interpretare i punteggi con cautela. Un singolo punteggio numerico spesso viene interpretato come se fosse preciso e non comportasse errori. Ad esempio, se si riporta che Alice ha un QI totale di 113, i suoi genitori potrebbero interpretarlo come se il QI di Alice fosse esattamente 113. Anche usando test di QI di alta qualità, i punteggi ottenuti contengono un certo grado di errore e il SEM e gli intervalli di confidenza ci aiutano a illustrare questo.

### Problema nel Calcolare l'Intervallo di Confidenza

Un problema potenziale con l'approccio descritto sopra è che non conosciamo il vero punteggio dell'esaminato, ma solo il punteggio osservato. È comune usare il SEM per stabilire intervalli di confidenza intorno ai punteggi ottenuti. Tuttavia, è importante sottolineare che questa pratica non è corretta {cite:p}`charter1996revisiting`. 

> In spite of {cite:t}`dudek1979continuing`'s reminder that the SEM should not be used to construct confidence intervals, many test manuals, computer-scoring programs, and texts in psychology and education continue to do so. Because authors of many textbooks and manuals make these errors, it is understandable that those who learned from and look to these sources for guidance also make these errors. In summary, the SEM should not be used to construct confidence intervals for test scores (p. 1141).

È invece possibile costruire gli intervalli di confidenza basati su punteggi veri stimati e sull'errore standard della stima (SEE). Questo approccio verrà descritto nel prossimo capitolo.

## Considerazioni conclusive

Nel contesto della CTT, le stime di affidabilità si rivelano uno strumento fondamentale per valutare la coerenza dei test. Tuttavia, quando si affrontano decisioni relative al singolo individuo, come ad esempio determinare se un candidato supera un esame, diventa più vantaggioso fare riferimento all'errore standard di misurazione (SEM). Il SEM rende evidente quanto i punteggi di un test siano suscettibili di fluttuazioni casuali se lo stesso test venisse ripetuto più volte dallo stesso esaminando. In generale, un SEM più ridotto corrisponde a un intervallo di fluttuazioni casuali più stretto. Ciò implica che, grazie a un SEM più basso, i punteggi rifletteranno in modo più coerente le vere capacità dell'esaminando.

## Session Info

In [2]:
sessionInfo()

R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] C

time zone: Europe/Rome
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] psychometric_2.4  multilevel_2.7    MASS_7.3-60.0.1   nlme_3.1-164     
 [5] ggokabeito_0.1.0  viridis_0.6.5     viridisLite_0.4.2 ggpubr_0.6.0     
 [9] ggExtra_0.10.1    bayesplot_1.11.1  gridExtra_2.3     patchwork_1.2.0  
[13] semTools_0.5-6    semPlot_1.1.6     lavaan_0.6-17     psych_2.4.1      
[17] scales_1.3.0      markdown_1.12     knitr_1.45        lubridate_1.9.3  
[21] forcats_1.0.0     stringr_1.5.1     dplyr_1.1.4       purrr_1.0.2      
[25] readr_2.